interfaces/builtin/mir.go

```diff
@@ -0,0 +1,179 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (c) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more dtails.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+        "bytes"
+)
+
+var mirPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as the Mir server. Reserved because this
+# gives privileged access to the system.
+# Usage: reserved
+
+capability dac_override,
+capability sys_tty_config,
+capability sys_admin,
+
+unix (receive, send) type=seqpacket addr=none,
+/dev/dri/card0 rw,
+/dev/shm/\#* rw,
+/dev/tty* wr,
+/run/udev/data/* r,
```

Comment:
When I remove /run/udev/data/* r, the server will receive these denials

Aug  4 20:19:13 localhost ubuntu-core-launcher[1499]: [2016-08-04 20:19:13.694297] mirplatform: Found graphics driver: mir:mesa-x11 (version 0.23.5)
Aug  4 20:19:13 localhost ubuntu-core-launcher[1499]: Unknown command line options: --vt 1
Aug  4 20:19:13 localhost kernel: [  589.117731] audit: type=1400 audit(1470341953.690:15): apparmor="DENIED" operation="open" profile="snap.mir-server.mir-server" name="/run/udev/data/+drm:card0-Virtual-1" pid=1501 comm="mir-server" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
Aug  4 20:19:13 localhost kernel: [  589.117859] audit: type=1400 audit(1470341953.690:16): apparmor="DENIED" operation="open" profile="snap.mir-server.mir-server" name="/run/udev/data/+drm:card0-Virtual-2" pid=1501 comm="mir-server" requested_mask="r" denied_mask="r" fsuid=0 ouid=0
Aug  4 20:19:13 localhost kernel: [  589.117948] audit: type=1400 audit(1470341953.690:17): apparmor="DENIED" operation="open" profile="snap.mir-server.mir-server" name="/run/udev/data/+drm:card0-Virtual-3" pid=1501 comm="mir-server" requested_mask="r" denied_mask="r" fsuid=0 ouid=0

So is there some alternative?
note, i already have /dev/dri/card0 rw, in the list